src/OpenLoco/src/Windows/MapWindow.cpp

```diff
@@ -1508,11 +1510,92 @@ namespace OpenLoco::Ui::Windows::MapWindow
         }
     }
 
-    // 0x0046CED0
-    static void sub_46CED0()
+    // 0x00478265
+    static void sub_478265(int8_t* buffer)
     {
         registers regs;
-        call(0x0046CED0, regs);
+        regs.edi = X86Pointer(buffer);
+        call(0x00478265, regs);
+    }
+
+    // 0x0046CED0
+    static void assignRouteColours()
+    {
+        uint32_t availableColours = 0x7FFFFFFF;
+
+        // First, assign water colour
+        {
+            auto waterObj = ObjectManager::get<WaterObject>();
+
+            auto waterPixel = Gfx::getG1Element(waterObj->mapPixelImage)->offset[0];
+            availableColours = checkIndustryColours(waterPixel, availableColours);
+
+            waterPixel = Gfx::getG1Element(waterObj->mapPixelImage)->offset[1];
+            availableColours = checkIndustryColours(waterPixel, availableColours);
+        }
+
+        // Then, assign surface texture colours
+        for (auto i = 0U; i < ObjectManager::getMaxObjects(ObjectType::land); i++)
+        {
+            auto landObj = ObjectManager::get<LandObject>(i);
+            if (landObj == nullptr)
+                continue;
+
+            auto landPixel = Gfx::getG1Element(landObj->mapPixelImage)->offset[0];
+            availableColours = checkIndustryColours(landPixel, availableColours);
+
+            landPixel = Gfx::getG1Element(landObj->mapPixelImage)->offset[1];
+            availableColours = checkIndustryColours(landPixel, availableColours);
+        }
+
+        availableColours = checkIndustryColours(PaletteIndex::index_3C, availableColours);
+        availableColours = checkIndustryColours(PaletteIndex::index_BA, availableColours);
+        availableColours = checkIndustryColours(PaletteIndex::index_D3, availableColours);
+        availableColours = checkIndustryColours(PaletteIndex::index_8B, availableColours);
+        availableColours = checkIndustryColours(PaletteIndex::index_0A, availableColours);
+        availableColours = checkIndustryColours(PaletteIndex::index_15, availableColours);
+
+        auto availableTracks = CompanyManager::getPlayerCompany()->getAvailableRailTracks();
+
+        std::array<int8_t, 100> buffer = {};
+        assert(std::size(buffer) >= std::size(availableTracks));
+
+        std::copy(std::begin(availableTracks), std::end(availableTracks), std::begin(buffer));
+        buffer[availableTracks.size()] = std::numeric_limits<uint8_t>::max();
+
+        auto* nextEl = &buffer[availableTracks.size()];
+        sub_478265(nextEl);
```

Comment:
Technically this function should also have updatingCompanyId set to player company for the duration as well but i really don't think that is needed.

Reply:
Yeah, I took that as a means to accommodate the function that is no longer necessary in our codebase.

Reply:
sub_478265 checks for unlocked vehicles using updatingCompanyId, is it assigned to the player at this point or why don't we need it?

Reply:
That would be probably best or we assert that the updating company id is that of a player.

Reply:
Following up in #2094